upgrade(rspec-rails): to 6.0.1 (was 5.1.2)

casaper opened the pull request:
The rspec-rails changelog between [5.1.0](https://github.com/rspec/rspec-rails/blob/main/Changelog.md/#510--2022-01-26) and [6.0.1](https://github.com/rspec/rspec-rails/blob/main/Changelog.md/#601--2022-10-18) (in the [rspec-rails CHANGELOG](https://github.com/rspec/rspec-rails/blob/main/Changelog.md)) only note the drop of support for Rails below 6.1 and Ruby below 2.5. Both should be fine with Mastodon, AFAIS.

Else there seems to be a lot of fixes and improvements, so I think the change probably makes sense

---

[fix(search_service_spec): expected keyword arguments in ruby 3.0](https://github.com/mastodon/mastodon/pull/22952/commits/04113974a71f9a23122815fe8d083ce271177faa)

Only in Ruby 3.0 the spec failed with:

```
 expected: ("tag", 10, 0, {:exclude_unreviewed=>nil}) (keyword arguments)
              got: ("tag", 10, 0, {:exclude_unreviewed=>nil}) (options hash)
```

---

[update(ruby): to 3.0.5 (was 3.0.4)](https://github.com/mastodon/mastodon/pull/22952/commits/04be8209018a00acf7608062317c5c87442b4f43)

Because of warning when running specs:
```
warning: parser/current is loading parser/ruby30, which recognizes 3.0.5-compliant syntax, but you are running 3.0.4.
```

---

[fix(rubocop_cache): CacheRootDirectory not working with editor](https://github.com/mastodon/mastodon/pull/22952/commits/fca85ba5a78661ad8b8ea2d966abce8f44036089)

For me this option causes my editor to produce rubocop cache directories
in the same directory as every file it opens, instead of in the project
root directorys `tmp/`.

ClearlyClaire commented:
Interestingly it causes/reveal a test failure with keyword args VS options hash on Ruby 3.0

casaper commented:
> Interestingly it causes/reveal a test failure with keyword args VS options hash on Ruby 3.0.
> 
> The code uses an option hash, the spec uses keywords. I'd be fine with it being fixed either way.

I'll look into this soon. Perhaps tomorrow.
Thx 👍

casaper committed:
upgrade(rspec-rails): to 6.0.1 (was 5.1.2)

casaper committed:
fix(search_service_spec): expected keyword arguments in ruby 3.0

Only in Ruby 3.0 the spec failed with:

```
 expected: ("tag", 10, 0, {:exclude_unreviewed=>nil}) (keyword arguments)
              got: ("tag", 10, 0, {:exclude_unreviewed=>nil}) (options hash)
```

casaper commented:
@ClearlyClaire 

> Interestingly it causes/reveal a test failure with keyword args VS options hash on Ruby 3.0.
> 
> The code uses an option hash, the spec uses keywords. I'd be fine with it being fixed either way.

I've got that fixed now. Was actually very simple to do. 

[fix(search_service_spec): expected keyword arguments in ruby 3.0](https://github.com/mastodon/mastodon/pull/22952/commits/04113974a71f9a23122815fe8d083ce271177faa)

Also had to do this one:

[update(ruby): to 3.0.5 (was 3.0.4)](https://github.com/mastodon/mastodon/pull/22952/commits/04be8209018a00acf7608062317c5c87442b4f43)

Because of warning when running specs:
```
warning: parser/current is loading parser/ruby30, which recognizes 3.0.5-compliant syntax, but you are running 3.0.4.
```

---

Now this seems ready for review to me.

ClearlyClaire reviewed (commented):
Can you rebase against `main`? Also I would not include `CacheRootDirectory` or ruby version changes in this PR as it is unrelated. Otherwise, LGTM.

nschonni commented on .rubocop.yml:
```diff
@@ -9,7 +9,6 @@ AllCops:
   DisplayStyleGuide: true
   ExtraDetails: true
   UseCache: true
```
Other thing to undo
```suggestion
  UseCache: true
  CacheRootDirectory: tmp
```

casaper replied:
https://github.com/mastodon/mastodon/pull/22952/commits/b4a37cf005cf0e8f8f093f9b447ff9c82ba5b0de

nschonni replied:
> Can you rebase against main? Also I would not include CacheRootDirectory or ruby version changes in this PR as it is unrelated. Otherwise, LGTM.

casaper replied:
I did, 10 minutes ago

casaper replied:
ah, I didn't parse that in Clairs comment. Sorry.

casaper replied:
removed the commit then.
